beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -807,15 +807,16 @@ private SszList<SignedValidatorRegistration> getApplicableValidatorRegistrations
                 signedValidatorRegistration -> {
                   final BLSPublicKey validatorIdentifier =
                       signedValidatorRegistration.getMessage().getPublicKey();
-                  final ValidatorStatus validatorStatus =
-                      validatorStatuses.get(validatorIdentifier);
-                  final boolean hasExited = validatorHasExited(validatorStatus);
-                  if (hasExited) {
+                  final boolean unknownOrHasExited =
+                      Optional.ofNullable(validatorStatuses.get(validatorIdentifier))
+                          .map(this::validatorHasExited)
+                          .orElse(true);
```

Comment:
I would suggest doing all the eligibility checks (has exited OR is unknown) in a separate method (probably rename `validatorHasExited` and do it there)... And rename the `unknownOrHasExited` to something like `isEligible`...